src/Lykke.AzureQueueIntegration/Publisher/AzureQueuePublisher.cs

```diff
@@ -56,20 +57,31 @@ public AzureQueuePublisher<TModel> SetSerializer(IAzureQueueSerializer<TModel> s
             if (_cloudQueue == null)
                 _cloudQueue = await _settings.GetQueueAsync();
 
-            do
+            QueueWithConfirmation<TModel>.QueueItem message = null;
+            try
             {
-                var message = _queue.Dequeue();
-
-                if (message == null)
+                do
                 {
-                    break;
-                }
-
-                var dataToQueue = _serializer.Serialize(message.Item);
-                await _cloudQueue.AddMessageAsync(new CloudQueueMessage(dataToQueue));
-                message.Compliete();
-
-            } while (true);
+                    message = _queue.Dequeue();
+
+                    if (message == null)
+                    {
+                        break;
+                    }
+
+                    var dataToQueue = _serializer.Serialize(message.Item);
+                    await _cloudQueue.AddMessageAsync(new CloudQueueMessage(dataToQueue));
+                    message.Compliete();
+                } while (true);
+            }
+            catch (Exception ex)
+            {
+                Log?.WriteError(
+                    $"{GetComponentName()}:{_settings.QueueName}",
+                    message == null || message.Item == null ? string.Empty : message.Item.ToJson(),
```

Comment:
```message == null || message.Item == null``` could be replaced with ```message?.Item == null```

Reply:
Can't apply ? to generic parameter TModel, cause it's not specified as class